ui/src/App.js

```diff
@@ -354,6 +357,28 @@ class App extends Component {
     return newValueList;
   }
 
+  removeFacet(facetValue) {
+    this.state.facets.delete(facetValue);
+    this.state.selectedFacetValues.delete(facetValue);
+    let extraFacetEsFieldNames = this.removeFacetValue(
```

Comment:
I see now that removeFacetValue isn't needed. We can just use [Array.filter()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter).

```
    this.setState({
￼       extraFacetEsFieldNames: extraFacetEsFieldNames.filter(n => n != facetValue),
￼     });
```

Do you mind deleting removeFacetValue()?